### PR TITLE
15 carousel

### PR DESCRIPTION
**Pull request by nrgix2**
# Feature/Carousel

>_Assigne_ : 
>Antoine MOUSSET

> _Task description :_
> This pull request enable to create a carousel with 2 props (title & content)

## Features

- [x] Create Carousel
- [x] Add props

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
     "core-js": "^3.8.3",
     "node-sass-magic-importer": "^5.3.2",
     "style-inject": "^0.3.0",
-    "vue": "^3.2.13"
+    "vue": "^3.2.13",
+    "vueperslides": "^3.3.2"
```
Comment (ChxGuillaume) [inline package.json:16]: remove this

File: src/index.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
-import HelloWorld from './components/HelloWorld/HelloWorld.vue';
+import CarouselVue from './components/Carousel/CarouselVue.vue';
 
 export default {
   install(Vue) {
-    Vue.component('HelloWorld', HelloWorld);
+    Vue.component('CarouselVue', CarouselVue);
```
Comment (ChxGuillaume) [inline src/index.js:5]: don't remove the hello world from exported components

**Review by ChxGuillaume — CHANGES_REQUESTED** [review]
Please make it your own.

File: package.json [hunk 3]
```diff
@@ -29,6 +29,9 @@
     "lib:publish:major": "npm version major && npm run lib:publish"
   },
   "dependencies": {
+    "core-js": "^3.8.3",
+    "node-sass-magic-importer": "^5.3.2",
+    "style-inject": "^0.3.0",
```
Comment (ChxGuillaume) [inline package.json:34]: ???

**Review by ChxGuillaume — CHANGES_REQUESTED** [review]
Missing caourssel item

**Comment by ChxGuillaume** [comment]
https://vuetifyjs.com/en/components/carousels/

File: src/components/QCarousel/QCarousel.vue [hunk 4]
```diff
@@ -0,0 +1,45 @@
+<template>
+  <div>
+    <button class="button" @click="prev()">Aller à gauche</button>
+    <button class="button" @click="next">Aller à droite</button>
+    <slot> </slot>
```
Comment (MrZyr0) [inline src/components/QCarousel/QCarousel.vue:5]:
```suggestion
    <slot />
```

File: src/components/QCarousel/QCarouselItem.vue [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<template>
+  <div class="card">
+    <slot></slot>
```
Comment (MrZyr0) [inline src/components/QCarousel/QCarouselItem.vue:3]:
```suggestion
    <slot />
```

**Review by MrZyr0 — DISMISSED** [review]

File: src/components/QCarousel/QCarousel.vue [hunk 6]
```diff
@@ -0,0 +1,102 @@
+<template>
+  <div class="q-carousel">
+    <button class="button" @click="selectItemFromIndex(state.selectedIndex - 1)">Aller à gauche</button>
+    <button class="button" @click="selectItemFromIndex(state.selectedIndex + 1)">Aller à droite</button>
```
Comment (ChxGuillaume) [inline src/components/QCarousel/QCarousel.vue:4]: why forcing it ?

File: src/components/QCarousel/QCarousel.vue [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+  children.forEach((item, index) => {
+    item.setAttribute('data-carousel-index', index);
+  });
+
+  setItemsClasses()
+});
+
+watch(() => state.selectedIndex, () => {
+
+})
```
Comment (ChxGuillaume) [inline src/components/QCarousel/QCarousel.vue:64]: ??????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????

File: src/components/QCarousel/QCarousel.vue [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+  left: 0;
+}
+
+.q-carousel-item--right {
+  left: 100%;
+}
+
+.q-carousel-item--left {
+  left: -100%;
+}
```
Comment (ChxGuillaume) [inline src/components/QCarousel/QCarousel.vue:101]: tailwind?

**Review by ChxGuillaume — CHANGES_REQUESTED** [review]

**Review by ChxGuillaume — APPROVED** [review]

**Pull request merged**